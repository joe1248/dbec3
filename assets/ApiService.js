import ApiServiceHelper from './js/lib/ApiServiceHelper';

class ApiService {

    constructor() {
        this.ApiServiceHelper = ApiServiceHelper;
    }

    getConnection(id, callback) {
        this.ApiServiceHelper.get(`http://api.local.dbec3.com/connection/` + id, callback);
    }

    saveConnection(connection, callback) {
        const connectionId = connection && connection.db_id || null;
        if (connectionId) {
            this.ApiServiceHelper.patch('http://api.local.dbec3.com/connection/edit', connection, callback);
        } else {
            this.ApiServiceHelper.post('http://api.local.dbec3.com/connection/new', connection, callback);
        }
    }
}

export default new ApiService;