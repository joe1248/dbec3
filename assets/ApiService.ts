import ApiServiceHelper from './js/lib/ApiServiceHelper';

class ApiService {
    ApiServiceHelper: any;

    constructor() {
        this.ApiServiceHelper = ApiServiceHelper;
    }

    getUserDatabaseConnections(callback: Function) {
        this.ApiServiceHelper.get(`/connections`, callback);
    }

    getConnection(id: Number, callback: Function) {
        this.ApiServiceHelper.get(`/connection/` + id, callback);
    }

    saveConnection(connection: UserDbConnection, callback: Function) {
        const connectionId = connection && connection.db_id || null;
        if (connectionId) {
            this.ApiServiceHelper.patch('/connection/edit', connection, callback);
        } else {
            this.ApiServiceHelper.post('/connection/new', connection, callback);
        }
    }

    deleteConnection(connectionId: Number, callback: Function) {
        if (connectionId) {
            this.ApiServiceHelper.erase('/connection/' + connectionId, callback);
        }
    }
}

export default new ApiService;