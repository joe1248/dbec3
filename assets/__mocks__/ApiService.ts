class ApiService {
    forceFailsGetUserDatabaseConnections: boolean = false;

    constructor() {
    }

    getConnection(id: Number, callback: Function) {
        if (id > 0) {
            callback(null, {
                db_connection_disabled: false,
                db_connection_name: 'bingo_test_db_connection_name',
                db_id: id,
                db_pass_word: 'db_password_111',
                db_port_number: '3333',
                db_selected_ftp_id: 777,
                db_url_host: 'db_host',
                db_user_name: 'db_user',
                ftp_connection_name: 'ssh_name',
                ftp_pass_word: 'ssh_pass',
                ftp_port_number: '22',
                ftp_url_host: 'ssh_host',
                ftp_user_name: 'ssh_user',
                select_db_protocol: 'over_ssh'
            });
            return;
        }
        callback('Error on purpose loading connection data', {});
    }

    saveConnection(connection: UserDbConnection, callback: Function) {
        if (connection && connection.db_id >= 0) {
            callback(null, connection);
            return;
        }
        callback('Error on purpose saving connection data', {});
    }

    deleteConnection(connectionId: Number, callback: Function) {
        if (connectionId >= 0) {
            callback(null);
            return;
        }
        callback('Error on purpose deleting connection data');
    }

    getUserDatabaseConnections(callback: Function) {
        if (this.forceFailsGetUserDatabaseConnections) {
            callback('Error on purpose getting UserDatabaseConnections.');
            return;
        }
        //console.log("\n\n\n\n\n-------------------------this\n\n");
        //console.log(this);
        callback(null, [
            {
                db_connection_disabled: false,
                db_connection_name: 'nice_test_db_connection_name',
                db_id: 123
            },
            {
                db_connection_disabled: true,
                db_connection_name: 'nice_too_test_db_connection_name',
                db_id: 456
            }
        ]);
    }
}
export default new ApiService;