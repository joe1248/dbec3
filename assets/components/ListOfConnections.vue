<template>
<div class="centered" id="app">
    <div style="width:100%;" class="centered">
        <br>
        <router-link :to="{name: 'NewConnection' }" tag="button">Add New DB Server</router-link>
        <br><br>
    </div>
    <fieldset class="fieldset one_line">
        <legend class="legend">Your DB Servers.</legend>
        <ul v-if="error">
            {{error}}
        </ul>

        <table style="width:400px;" v-if="!loading && connections && connections.length">
            <tr v-for="connection of connections">
                <td class="centered" style="width:20px;">
                    <div class="connection_statuses">&nbsp;</div>
                </td>
                <td style="text-indent:20px;">{{connection.db_connection_name}}</td>
                <td class="centered" id="td_button_connection_sql_id" style="width:55px;">&nbsp;	</td>
                <!--<td><input type="button" value="Enable"	html_connection_id="" html_connection_label="" class="small-button button_connection_enable"></td>-->
                <td><router-link :to="{name: 'EditConnection', params: { id: connection.db_id } }" tag="button" class="small-button">Edit</router-link></td>
                <td><button @click="deleteConnection(connection)" class="small-button deleteConnectionClass">Delete</button></td>
                <!--<td><input type="button" value="Clone"		html_connection_id="" html_connection_label="" class="small-button button_connection_clone"></td>-->
            </tr>
        </table>
        <Loader v-else="!loading"/>
    </fieldset>
</div>
</template>

<script lang="ts">
import ApiService from './../ApiService';
import Styling from './../js/lib/Styling.js';
import Loader from './Loader';

export default {
    data () {
        return {
            loading: false,
            connections: [],
            error: null
        }
    },

    created() {
        this.fetchData()
    },

    methods: {
        fetchData() {
            this.loading = true;
            ApiService.getUserDatabaseConnections((err: string, data: object) => {
                this.loading = false;
                if (err) {
                    this.error = err.toString();
                    return ;
                }
                this.connections = data;
                this.decorateUi();
            });
        },

        decorateUi() {
            this.$nextTick(function () {
                Styling.resetStyles();
            });
        },

        deleteConnection(connection) {
            ApiService.deleteConnection(connection.db_id, (err) => {
                if (err) {
                    this.error = err.toString();
                    return;
                }
                this.fetchData();
            });
        }

    }, // End of methods
    components: {
        // <my-component> will only be available in parent's template
        'Loader': Loader
    }
}
</script>