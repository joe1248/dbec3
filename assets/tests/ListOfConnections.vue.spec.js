import { mount, shallow } from '@vue/test-utils' // see https://vue-test-utils.vuejs.org
import ListOfConnections from './../components/ListOfConnections'
import { createRenderer } from 'vue-server-renderer'

jest.mock('./../ApiService');
import ApiService from './../ApiService';

beforeEach(() => {
    // Clear all instances and calls to constructor and all methods:
    //ApiServiceMock.mockReset();
});

describe('ListOfConnectionsComponent', () => {

    test('is a Vue instance', () => {
        const wrapper = mount(ListOfConnections);
        expect(wrapper.isVueInstance()).toBeTruthy()
    });

    test('should fails on purpose', function () {
        ApiService.forceFailsGetUserDatabaseConnections = true;
        const wrapper = shallow(ListOfConnections);
        // Match Object
        expect(wrapper.vm._data).toEqual({
            loading: false,
            connections: [],
            error: 'Error on purpose getting UserDatabaseConnections.'
        });
        const renderer = createRenderer();
        renderer.renderToString(wrapper.vm, (err, str) => {
            if (err) throw new Error(err.toString());
            expect(str).toMatchSnapshot()
        });
        ApiService.forceFailsGetUserDatabaseConnections = false;
    });

    test('should fetch SUCCESS without any props', function () {
        const wrapper = shallow(ListOfConnections);
        // Match Object
        expect(wrapper.vm._data).toEqual({
            loading: false,
            connections: [
                {
                    db_connection_disabled: false,
                    db_connection_name: 'nice_test_db_connection_name',
                    db_id: 123
                },
                {
                    db_connection_disabled: true,
                    db_connection_name: 'nice_too_test_db_connection_name',
                    db_id: 456
                }
            ],
            error: null
        });
        const renderer = createRenderer();
        renderer.renderToString(wrapper.vm, (err, str) => {
            if (err) throw new Error(err.toString());
            expect(str).toMatchSnapshot()
        });
    });

    test('should delete success', function () {
        const wrapper = shallow(ListOfConnections);
        const deleteButton = wrapper.find('.deleteConnectionClass');
        deleteButton.trigger('click');

        expect(wrapper.vm._data).toEqual({
            loading: false,
            connections: [
                {
                    db_connection_disabled: false,
                    db_connection_name: 'nice_test_db_connection_name',
                    db_id: 123
                },
                {
                    db_connection_disabled: true,
                    db_connection_name: 'nice_too_test_db_connection_name',
                    db_id: 456
                }
            ],
            error: null
        });

        const renderer = createRenderer();
        renderer.renderToString(wrapper.vm, (err, str) => {
            if (err) throw new Error(err.toString());
            expect(str).toMatchSnapshot()
        })
    });

    test('should delete fails', function () {
        const wrapper = shallow(ListOfConnections);
        wrapper.vm.deleteConnection(-1);
        expect(wrapper.vm._data).toEqual({
            loading: false,
            connections: [
                {
                    db_connection_disabled: false,
                    db_connection_name: 'nice_test_db_connection_name',
                    db_id: 123
                },
                {
                    db_connection_disabled: true,
                    db_connection_name: 'nice_too_test_db_connection_name',
                    db_id: 456
                }
            ],
            error: 'Error on purpose deleting connection data'
        });
    });
});