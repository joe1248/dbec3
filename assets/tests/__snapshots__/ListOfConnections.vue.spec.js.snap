// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfConnectionsComponent should delete success 1`] = `
<div id="app" class="centered">
    <div class="centered" style="width:100%;">
        <br>
        <router-link to="[object Object]" tag="button">Add New DB Server</router-link>
        <br>
        <br>
    </div>
    <fieldset class="fieldset one_line">
        <legend class="legend">Your DB Servers.</legend>
        <!---->
        <table style="width:400px;">
            <tr>
                <td class="centered" style="width:20px;">
                    <div class="connection_statuses"> </div>
                </td>
                <td style="text-indent:20px;">nice_test_db_connection_name</td>
                <td id="td_button_connection_sql_id" class="centered" style="width:55px;">  </td>
                <td>
                    <router-link to="[object Object]" tag="button" class="small-button">Edit</router-link>
                </td>
                <td>
                    <button class="small-button deleteConnectionClass">Delete</button>
                </td>
            </tr>
            <tr>
                <td class="centered" style="width:20px;">
                    <div class="connection_statuses"> </div>
                </td>
                <td style="text-indent:20px;">nice_too_test_db_connection_name</td>
                <td id="td_button_connection_sql_id" class="centered" style="width:55px;">  </td>
                <td>
                    <router-link to="[object Object]" tag="button" class="small-button">Edit</router-link>
                </td>
                <td>
                    <button class="small-button deleteConnectionClass">Delete</button>
                </td>
            </tr>
        </table>
    </fieldset>
</div>
`;

exports[`ListOfConnectionsComponent should fails on purpose 1`] = `
<div id="app" class="centered">
    <div class="centered" style="width:100%;">
        <br>
        <router-link to="[object Object]" tag="button">Add New DB Server</router-link>
        <br>
        <br>
    </div>
    <fieldset class="fieldset one_line">
        <legend class="legend">Your DB Servers.</legend>
        <ul>
            Error on purpose getting UserDatabaseConnections.
        </ul>
        <!---->
    </fieldset>
</div>
`;

exports[`ListOfConnectionsComponent should fetch SUCCESS without any props 1`] = `
<div id="app" class="centered">
    <div class="centered" style="width:100%;">
        <br>
        <router-link to="[object Object]" tag="button">Add New DB Server</router-link>
        <br>
        <br>
    </div>
    <fieldset class="fieldset one_line">
        <legend class="legend">Your DB Servers.</legend>
        <!---->
        <table style="width:400px;">
            <tr>
                <td class="centered" style="width:20px;">
                    <div class="connection_statuses"> </div>
                </td>
                <td style="text-indent:20px;">nice_test_db_connection_name</td>
                <td id="td_button_connection_sql_id" class="centered" style="width:55px;">  </td>
                <td>
                    <router-link to="[object Object]" tag="button" class="small-button">Edit</router-link>
                </td>
                <td>
                    <button class="small-button deleteConnectionClass">Delete</button>
                </td>
            </tr>
            <tr>
                <td class="centered" style="width:20px;">
                    <div class="connection_statuses"> </div>
                </td>
                <td style="text-indent:20px;">nice_too_test_db_connection_name</td>
                <td id="td_button_connection_sql_id" class="centered" style="width:55px;">  </td>
                <td>
                    <router-link to="[object Object]" tag="button" class="small-button">Edit</router-link>
                </td>
                <td>
                    <button class="small-button deleteConnectionClass">Delete</button>
                </td>
            </tr>
        </table>
    </fieldset>
</div>
`;
